### DISCUSSION TITLE
Something wrong happen when building PSF

### DISCUSSION THREAD

**Discussion by Yanshengyu0311** [Astro-Sean/autophot#6] [Q&A] [answered]
+-------------------------------------------+
|Building PSF model using stars in the field|
+-------------------------------------------+

Could not fit background: All boxes contain > 4.9 (10 percent per box) masked pixels (or all are completely masked). Please check your data or increase "exclude_percentile" to allow more boxes to be included.
 
In background.remove_background line 117-125
            background = Background2D(image,
                                        box_size = box,
                                        mask = mask_array,
                                        filter_size = filter_size,
                                        sigma_clip = sigma_clip,
                                        bkg_estimator = SExtractorBackground(sigma_clip=sigma_clip),
                                        interpolator= BkgZoomInterpolator(order=6),
                                        exclude_percentile = 10
                                        )
I try to increase the exclude_percentile =50 or increase the order=6 of BkgZoomInterpolator, It still have this problem. I didn't use photutils before, so I need your  valuable suggestions, Thanks in advace!

**Comment by Astro-Sean** [comment, marked as answer]
Hi - so this error can come about when you can't fit a surface to the background around a star when there is "junk" in the enviroment.

For example in IR images (especially K-band) there can be correlation noise that source detection can pick up as sources. 

To fix this issue lets turn off surface fitting and use the more tradiational median from the annulus background 

Put this in before running autophot:

autophot_input['fitting']['remove_bkg_surface'] = False
autophot_input['fitting']['remove_bkg_local'] = True

**Reply by Yanshengyu0311** [reply]
Thanks, I got it! Fitting background use Surface fitting looks advanced. It's a nice process in the pipeline. But traditional background estimation seems more suitable for my data.
